packages/x-date-pickers/src/DateCalendar/DayCalendar.tsx

```diff
@@ -565,9 +565,7 @@ export function DayCalendar<TDate>(inProps: DayCalendarProps<TDate>) {
           </PickersCalendarWeekNumberLabel>
         )}
         {getWeekdays(utils, now).map((weekday, i) => {
-          // regression-prevention:
-          // since 'weekdayShort' now always returns an abbreviated form we slice the first 2 letters from it.
-          const day = utils.format(weekday, 'weekdayShort').slice(0, 2);
```

Comment:
The `slice` was causing the discrepancy between behavior before v6.16.0. 🤔

Reply:
oh, yeah ... that's a blunder I made 😮‍💨 ... Sry! We do this in the default `dayOfWeekFormatter` in a similar way. I guess I just wanted to pass the first 2 chars since that was the previous behaviour

Reply:
> oh, yeah ... that's a blunder I made 😮‍💨 ... Sry! We do this in the default `dayOfWeekFormatter` in a similar way. I guess I just wanted to pass the first 2 chars since that was the previous behaviour

Was it? 🤔 
I believe that we were not doing any preformatting of the resulting item from the `adapter.getWeekdays` before. 🤷

Reply:
well, previously we just returned a formatted string from the adapter method `getWeekDays`. But the format Token was hard coded. Maybe thats why I thought it's always the same